Code/Framework/AzCore/AzCore/Name/Name.cpp

```diff
@@ -44,39 +46,51 @@ namespace AZ
         *this = rhs;
     }
 
+    Name::Name(NameRef name)
+        : m_data(AZStd::move(name))
+        , m_hash(name->GetHash())
+        , m_view(name->GetName())
+    {
+    }
+
+    Name Name::FromStringLiteral(AZStd::string_view name)
+    {
+        Name literalName;
+        literalName.SetNameLiteral(name);
+        return literalName;
+    }
+
     Name& Name::operator=(const Name& rhs)
     {
-        m_data = rhs.m_data;
-        m_hash = rhs.m_hash;
-        if (!rhs.m_view.empty())
+        if (rhs.m_supportsDeferredLoad && rhs.m_data == nullptr)
         {
-            m_view = rhs.m_view;
+            m_hash = rhs.m_hash;
+            m_data = rhs.m_data;
+            SetNameLiteral(rhs.GetStringView());
```

Comment:
I'm starting to think we might need extra care in copying m_view. If you copy a Name across dll boundaries, you could end up with a Name in one module pointing to a string literal in another module. What happens if we unload that other module? If the Name's text is stored in a string literal, but not in the NameDictionary yet, and the Name is copied, then we might need to _copy_ the text too. In that case we would need to allocate the data in system memory because allocators aren't available during static initialization.

Reply:
This can only occur if the name literal gets copied before the name dictionary is initialized (m_data is null), which will be before module initialization time because literals are loaded at module creation time, so I can't think of a scenario where you could accidentally copy the literal reference across module boundaries